pkg/vsphere/tasks/waiter.go

```diff
@@ -89,11 +89,26 @@ func WaitForResult(ctx context.Context, f func(context.Context) (Task, error)) (
 	}
 }
 
-func isTaskInProgress(err error) bool {
+const (
+	vimFault  = "vim"
+	soapFault = "soap"
+	taskFault = "task"
+)
+
+// isRetryErrors will return true for vSphere errors, which can be fixed by retry.
+// Currently the error includes TaskInProgress, NetworkDisruptedAndConfigRolledBack and InvalidArgument
+// The reason to retry for these errors is because sometimes vSphere needs longer time to finish sync up inside
+func isRetryErrors(err error) bool {
 	if soap.IsSoapFault(err) {
 		switch f := soap.ToSoapFault(err).VimFault().(type) {
 		case types.TaskInProgress:
 			return true
+		case *types.NetworkDisruptedAndConfigRolledBack:
+			logExpectedFault(soapFault, f)
+			return true
+		case *types.InvalidArgument:
```

Comment:
hmm.. I don't think we're masking bugs. Even with retry, the bug will still fail after a few times retry.
No matter how, our bug needs to be fixed, no matter retry or not. But if we move the retry logic to somewhere else, that means anyone powerOn VM needs to check if that's run on vSAN, and needs to take care of the retry.
I personally dislike to spread the workaround code for vSAN bug, or even not a bug (just test env issue) in everywhere.
So if it doesn't hurt, I'm fine to have some retry workaround in one central place, or we might just need to justify this should be fixed in vSAN. 😄